packages/gateway/src/errors.js

```diff
@@ -0,0 +1,25 @@
+export class InvalidIpfsPathError extends Error {
+  /**
+   * @param {string} cid
+   */
+  constructor(cid) {
+    super(`invalid ipfs path: invalid path "/ipfs/${cid}/"`)
+    this.name = 'InvalidIpfsPath'
```

Comment:
```suggestion
    this.name = 'InvalidIpfsPathError'
```

For consistency with `AbortError`:

> Note: When `abort()` is called, the `fetch()` promise rejects with an `Error` of type `DOMException`, with name `AbortError`.

https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort